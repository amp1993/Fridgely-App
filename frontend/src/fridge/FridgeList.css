
.empty-grocery-list-container{
    display: flex;
    justify-content: center;
    margin-top: 20px;
}
.item-buttons {
justify-content: center;
margin: -10px;
}