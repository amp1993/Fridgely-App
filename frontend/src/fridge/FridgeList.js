import React, { useEffect } from "react";
import { useNavigate } from "react-router-dom";
import FridgelyApi from "../api/api";
import FridgeItemCard from "./FridgeItemCard";
import NotAuthorized from "../common/NotAuthorize";
import "./FridgeList.css"


// FridgeList component to display fridge items or a message if the fridge is empty
const FridgeList = ({ fridgeItems, user, updateFridgeItems }) => {
    const navigate = useNavigate();


    useEffect(() => {
         // Fetch fridge items on component mount
        async function fetchFridgeItems() {
            try {
                const fridgeData = await FridgelyApi.getFridgeItems(user);
                // const data = fridgeData.items;
                if (fridgeData.length > 0) {
                    updateFridgeItems(fridgeData)

                }
                return
            } catch (error) {
                console.error('Error fetching recipes:', error);
            }
        }
        fetchFridgeItems();
    }, []);
    // Navigate to the page for adding a new item to the fridge
    const handleCick = ()=>{
        navigate( `/fridge/${user}`)
    }

   

    return (
        <>
            {user ? (
                fridgeItems.length > 0 ? (
                    <>
                        <FridgeItemCard fridgeItems={fridgeItems} user={user} updateFridgeItems={updateFridgeItems} />
                    </>
                ) : (
                    // Display message and button if the fridge is empty

                    <>
                        <div className="empty-grocery-list-container">
                            <p>Fridge is Empty</p>
                        </div>
                   <div className="item-buttons">
                   <button onClick={handleCick}>Add an Item</button>
                   </div>
                           
                   
                    </>
                )
            ) : (
                // Display NotAuthorized component if the user is not authorized

                <NotAuthorized />)}
        </>
    );

};

export default FridgeList;