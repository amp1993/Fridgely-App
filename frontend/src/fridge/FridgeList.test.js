import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import FridgeList from './FridgeList';

const mockFridgeItems = [
  { id: 1, productName: 'Apple', categoryName: 'Fruits', quantityInFridge: 5, unitOfMeasure: 'pcs' },
  { id: 2, productName: 'Carrot', categoryName: 'Vegetables', quantityInFridge: 10, unitOfMeasure: 'pcs' },
];

it('renders without crashing', function() {
  render(
    <BrowserRouter>
      <FridgeList fridgeItems={[]} updateFridgeItems={() => {}} user={null} />
    </BrowserRouter>
  );
});

it('matches snapshot', function() {
  const { asFragment } = render(
    <BrowserRouter>
      <FridgeList fridgeItems={[]} updateFridgeItems={() => {}} user={null} />
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it('displays the correct message when user is not logged in', function() {
  const { getByText } = render(
    <BrowserRouter>
      <FridgeList fridgeItems={[]} updateFridgeItems={() => {}} user={null} />
    </BrowserRouter>
  );

  expect(getByText('Please log in to access page.')).toBeInTheDocument();
});

it('displays the correct message when fridge is empty and user is logged in', function() {
  const { getByText } = render(
    <BrowserRouter>
      <FridgeList fridgeItems={[]} updateFridgeItems={() => {}} user="testuser" />
    </BrowserRouter>
  );

  expect(getByText('Fridge is Empty')).toBeInTheDocument();
  expect(getByText('Add an Item')).toBeInTheDocument();
});

