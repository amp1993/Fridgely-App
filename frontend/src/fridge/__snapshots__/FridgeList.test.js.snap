// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <h1>
      Please log in to access page.
    </h1>
  </div>
</DocumentFragment>
`;
